codecvt_utf8 removed in C++26

https://github.com/CLIUtils/CLI11/blob/063b2c911cea8bd4b908d6187c6a007ad320cb1a/include/CLI/Macros.hpp#L98-L104

Using llvm 19, in C++26 mode, can't compile

```
In file included from /Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/CLI.hpp:18:
In file included from /Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/Encoding.hpp:54:
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:68:17: error: no member named 'wstring_convert' in namespace 'std'
   68 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(str, str + str_size);
      |            ~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:68:38: error: no member named 'codecvt_utf8' in namespace 'std'
   68 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(str, str + str_size);
      |                                 ~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:68:58: error: expected '(' for function-style cast or type construction
   68 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(str, str + str_size);
      |                                                   ~~~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:68:61: error: expected expression
   68 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().to_bytes(str, str + str_size);
      |                                                             ^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:99:17: error: no member named 'wstring_convert' in namespace 'std'
   99 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(str, str + str_size);
      |            ~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:99:38: error: no member named 'codecvt_utf8' in namespace 'std'
   99 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(str, str + str_size);
      |                                 ~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:99:58: error: expected '(' for function-style cast or type construction
   99 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(str, str + str_size);
      |                                                   ~~~~~~~^
/Users/julien/.conan2/p/cli1176dd194db6a54/p/include/CLI/impl/Encoding_inl.hpp:99:61: error: expected expression
   99 |     return std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(str, str + str_size);
```

--- a/include/CLI/TypeTools.hpp
+++ b/include/CLI/TypeTools.hpp
@@ -544,7 +544,8 @@ template <typename T, std::size_t I>
 }
 
 /// Get the type size of the sum of type sizes for all the individual tuple types
-template <typename T> struct type_count<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
+template <typename T>
+struct type_count<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
     static constexpr int value{tuple_type_size<T, 0>()};
 };
 
@@ -593,7 +594,8 @@ template <typename T, std::size_t I>
 }
 
 /// Get the type size of the sum of type sizes for all the individual tuple types
-template <typename T> struct type_count_min<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
+template <typename T>
+struct type_count_min<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
     static constexpr int value{tuple_type_size_min<T, 0>()};
 };
 

--- a/include/CLI/TypeTools.hpp
+++ b/include/CLI/TypeTools.hpp
@@ -544,7 +544,7 @@ template <typename T, std::size_t I>
 }
 
 /// Get the type size of the sum of type sizes for all the individual tuple types
-template <typename T> struct type_count<T, typename std::enable_if<is_tuple_like<T>::value>::type> {
+template <typename T> struct type_count<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
     static constexpr int value{tuple_type_size<T, 0>()};
 };
 
@@ -593,7 +593,7 @@ template <typename T, std::size_t I>
 }
 
 /// Get the type size of the sum of type sizes for all the individual tuple types
-template <typename T> struct type_count_min<T, typename std::enable_if<is_tuple_like<T>::value>::type> {
+template <typename T> struct type_count_min<T, typename std::enable_if<is_tuple_like<T>::value && !is_complex<T>::value>::type> {
     static constexpr int value{tuple_type_size_min<T, 0>()};
 };
 

